modules/apps/asset/asset-service/src/main/java/com/liferay/asset/internal/util/AssetHelperImpl.java

```diff
@@ -554,8 +555,24 @@ private AssetSearcher _getAssetSearcher(
 			searchContext.setLike(true);
 		}
 
-		searchContext.setSorts(
-			_getSorts(assetEntryQuery, searchContext.getLocale()));
+		boolean use_new_sorts_from_portal_search_api = true;
```

Comment:
@arboliveira, are you suggesting me to create a config/portal property or this is only for guidance purpose?

Reply:
Sorry, only for guidance 😅 I recommend moving from `searchContext.setSorts(...)` to `searchRequestBuilder.sorts(...)` in all cases; and you will need to rewrite `getSorts` so it returns sorts using the technique illustrated by `BaseNestedFieldsSortTestCase.assertSort`. We only need to confirm the new approach with nested fields will work with Solr as well (cc/ @BryanEngler), otherwise configuration may be needed for Solr users to fall back to previous implementation of DDM field names and sorts.